controller/konnect/reconciler_generic.go

```diff
@@ -412,6 +409,13 @@ func (r *KonnectEntityReconciler[T, TEnt]) Reconcile(
 		return ctrl.Result{}, nil
 	}
 
+	// Handle type specific operations and stop reconciliation if needed.
+	// This can happen for instance when KongConsumer references credentials Secrets
```

Comment:
Does it mean that we will halt the reconciliation at this stage if a `KongConsumer` has multiple credential references and one is invalid (e.g., it doesn't exist)? Couldn't we populate the condition but keep on rolling to create the entity in Konnect and not behave using the "all-or-nothing" strategy? WDYT?

Reply:
> Does it mean that we will halt the reconciliation at this stage if a KongConsumer has multiple credential references and one is invalid (e.g., it doesn't exist)?

Correct. I can be persuaded on this but my thinking was that it'd be better to stop the reconciliation rather than create a consumer without the requested credentials.

Reply:
From a security standpoint, it shouldn't be a threat to create a consumer without credentials, thus my question. I don't have a strong opinion on that, to be fair. One potential problem we could hit is that if there was an already existing consumer, adding an invalid credential ref would block its updates until the credential references get fixed.

Reply:
I'm approving the PR and leaving it up to you to decide what is the right behavior here.

Reply:
> One potential problem we could hit is that if there was an already existing consumer, adding an invalid credential ref would block its updates until the credential references get fixed.

That's a valid concern. I'm actually thinking what should be the result of this:

- reconciliation: whether it should stop or not?
  - if it should stop (current status) I believe it should set both `CredentialSecretRefsValid` and `Programmed` to false. (current status of this PR)
  - if it should continue: I _think_ it should set `Programmed` to `True` and only `CredentialSecretRefsValid` to false. That can make this error hard to spot

I think I'm leaning towards leaving this as is but I'm open to have a discussion on this.

Reply:
I'm going to merge this PR to unblock #1259 but I'm happy to change this if we get to a consensus that this shouldn't stop the reconciliation.

Reply:
> I think I'm leaning towards leaving this as is but I'm open to have a discussion on this. My reasoning is that the resource has a bad state and should be fixed ( even though it might be considered usable in this half broken state).

I agree with this reasoning. 👍